src/System.Memory/tests/Performance/Perf.ReadOnlySequence.Slice.cs

```diff
@@ -0,0 +1,1069 @@
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySequence<byte> sliced = buffer;
+                        long sliceLen = sliced.Length;
+                        while (sliceLen > 0)
+                        {
+                            sliceLen -= 2 * offset;
+                            SequencePosition start = sliced.GetPosition(offset);
+                            sliced = sliced.Slice(start, sliceLen);
+                            localInt ^= sliced.Start.GetInteger();
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(10_000, 100)]
+        private static void Byte_MultiSegment_PosPos(int bufSize, int bufOffset)
+        {
+            var segment1 = new BufferSegment<byte>(new byte[bufSize / 10]);
+            BufferSegment<byte> segment2 = segment1;
+            for (int j = 0; j < 10; j++)
+                segment2 = segment2.Append(new byte[bufSize / 10]);
+            var buffer = new ReadOnlySequence<byte>(segment1, bufOffset, segment2, bufSize / 10 - bufOffset);
+            long offset = buffer.Length / 20;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySequence<byte> sliced = buffer;
+                        long sliceLen = sliced.Length;
+                        while (sliceLen > 0)
+                        {
+                            sliceLen -= offset;
+                            SequencePosition end = sliced.GetPosition(sliceLen);
+                            SequencePosition start = sliced.GetPosition(offset);
+                            sliceLen -= offset;
+                            sliced = sliced.Slice(start, end);
+                            localInt ^= sliced.Start.GetInteger();
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        #endregion
+
+        #region Byte_Empty
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void Byte_Empty()
+        {
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
```

Comment:
This doesn't look like a great performance test.What exactly is it measuring? Combining several API calls doesn't give us very useful data if the work is not a meaningful scenario. **I would remove this test all together.**

Reply:
I can split this test and next test to each Slice method type.

Reply:
I don't know how valuable the results of those tests would be (in general, the Empty/Default performance tests might be overkill). Splitting them would be better, but it would be fine to remove as well. We don't want to add too much noisy data hiding things that really matter.

Reply:
This tests are useful that they show time on empty sequence so we can see as slice is optimal on noт empty sequences. I added them to see that Slice it optimal on non empty sequences. It is like reference values. That is why I group them in one test. Because we do not need many reference values.

Reply:
I can remove them.